lib/kontena/actors/lb_supervisor.rb

```diff
@@ -0,0 +1,86 @@
+module Kontena::Actors
+  class LbSupervisor < Concurrent::Actor::RestartingContext
+    include Kontena::Logging
+
+    # @param [Message] msg
+    def on_message(msg)
+      type = msg.class
+
+      if msg.is_a?(Concurrent::ImmutableStruct::ActorMessage)
+        on_message_object(msg)
+      elsif msg.is_a?(Symbol)
+        on_message_symbol(msg)
+      else
+        warn "unknown message type: #{type}"
+        pass
+      end
+    end
+
+    def on_message_object(msg)
+      case msg.action
+      when :start
+        start
+      when :generate_config
+        generate_config(msg.value)
+      when :write_config
+        write_config(msg.value)
+      when :update_haproxy
+        update_haproxy
+      else
+        pass
+      end
+    end
+
+    def on_message_symbol(action)
+      case action
+      when :reset
+        info "got reset event"
+      else
+        info "unhandled message symbol: #{action}"
```

Comment:
Should this `pass` instead?

Reply:
Maybe, I just wasn't comfortable with crashing top-level supervisor if it gets unknown message.